demo/docbot/cluster_bootstrap.py

```diff
@@ -1,305 +1,311 @@
+from os import getenv
+import dotenv
+from opensearchpy import OpenSearch
+from docbot.util import opensearch_connection_builder, opensearch_compare_dictionaries, MLClient
 import requests
-import os, json, sys
+import os
+import json
+import sys
 sys.path.append('./demo/')
-from docbot.util import opensearch_connection_builder, opensearch_compare_dictionaries, MLClient
-from opensearchpy import OpenSearch
-
-import dotenv
-from os import getenv
 
 dotenv.load_dotenv()
-COHERE_KEY = getenv('COHERE_KEY')
-MODEL_STATE = {
-    "model_group_id": "",
-    "connector_id": "",
-    "model_id": ""
-}
-
-
-def initialize_cluster_settings(client: OpenSearch) -> None:
-    """
-    Args:
-        Client: OpenSearch
-    Returns:
-        returns None if cluster settings was created or is exists else raise Exception
-    """
-    current_settings = client._client.cluster.get_settings()
-
-    data = {
-        "plugins.ml_commons.allow_registering_model_via_url": True,
-        "plugins.ml_commons.only_run_on_ml_node": False,
-        "plugins.ml_commons.connector_access_control_enabled": True,
-        "plugins.ml_commons.model_access_control_enabled": True,
-        "plugins.ml_commons.memory_feature_enabled": True,
-        "plugins.ml_commons.rag_pipeline_feature_enabled": True,
-        "plugins.ml_commons.trusted_connector_endpoints_regex": [
-            "^https://runtime\\.sagemaker\\..*[a-z0-9-]\\.amazonaws\\.com/.*$",
-            "^https://api\\.openai\\.com/.*$",
-            "^https://api\\.cohere\\.ai/.*$"
-        ]
-    }
-
-    if 'persistent' in current_settings and opensearch_compare_dictionaries(data, current_settings['persistent']):
-        print ("Cluster settings are already initialized!")
-        return
-
-    response = client.cluster.put_settings(body={"persistent": data})
-
-    # Check if request was successful
-    if response and 'acknowledged' in response and response['acknowledged']:
-        print("Cluster settings initialized successfully!")
-    else:
-        raise Exception("Failed to initialize cluster settings.")
 
+class ClusterBootstrap:
+    def __init__(self, use_ssl=True):
+        self.Cohere_key = getenv('COHERE_KEY')
+        self.model_group_id = ""
+        self.connector_id = ""
+        self.embedding_model = ""
+        self.language_model = ""
+        self.client = opensearch_connection_builder(
+            ml_client=True, use_ssl=use_ssl)
+
+        self.initialize_cluster_settings()
+        self.initialize_model_group()
+        self.initialize_connector()
+        self.initialize_model("embed-english-v2.0", "Embedding Model")
+        self.initialize_model("command-nightly", "Cohere Command Model")
+        self.initialize_ingestion_pipeline()
+        self.initialize_index()
```

Comment:
For these lines lets prefix with a `_` so that others will know that they are for internal use.